diff --git a/tests/data/test4002 b/tests/data/test4002
new file mode 100644
index 0000000..1616b7b
--- /dev/null
+++ b/tests/data/test4002
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="US-ASCII"?>
+<testcase>
+<info>
+<keywords>
+HTTP
+HTTP GET
+z/OS tagging
+</keywords>
+</info>
+
+# Server-side
+<reply>
+<data crlf="headers">
+HTTP/1.1 200 OK
+Date: Tue, 09 Nov 2010 14:49:00 GMT
+Server: test-server/fake
+Content-Length: 6
+Connection: close
+Content-Type: text/plain
+
+-foo-
+</data>
+</reply>
+
+# Client-side
+<client>
+<server>
+http
+</server>
+<features>
+zos
+</features>
+<name>
+z/OS tagging: existing binary file stays binary
+</name>
+<precheck>
+rm -f %LOGDIR/curl%TESTNUMBER.out && touch %LOGDIR/curl%TESTNUMBER.out && chtag -b %LOGDIR/curl%TESTNUMBER.out
+</precheck>
+<command>
+http://%HOSTIP:%HTTPPORT/%TESTNUMBER
+</command>
+</client>
+
+# Verify data after the test has been "shot"
+<verify>
+<postcheck>
+ls -T %LOGDIR/curl%TESTNUMBER.out | grep "^b "
+</postcheck>
+<protocol crlf="headers">
+GET /%TESTNUMBER HTTP/1.1
+Host: %HOSTIP:%HTTPPORT
+User-Agent: curl/%VERSION
+Accept: */*
+
+</protocol>
+</verify>
+</testcase>
