diff --git i/src/tool_cb_wrt.c w/src/tool_cb_wrt.c
index 1b76565..d890b15 100644
--- i/src/tool_cb_wrt.c
+++ w/src/tool_cb_wrt.c
@@ -28,6 +28,13 @@
 #include "tool_cb_wrt.h"
 #include "tool_operate.h"
 
+#ifdef __MVS__
+#include <zos.h>
+#include <fcntl.h>
+#include <_Ccsid.h>
+
+#endif
+
 #ifdef _WIN32
 #define OPENMODE S_IREAD | S_IWRITE
 #else
@@ -99,6 +106,17 @@ bool tool_create_output_file(struct OutStruct *outs,
           curlx_strerror(errno, errbuf, sizeof(errbuf)));
     return FALSE;
   }
+
+#ifdef __MVS__
+  if(config->local_ccsid) {
+    int fd = fileno(file);
+    if(config->local_ccsid == 65535)
+      __chgfdccsid(fd, FT_BINARY);
+    else
+      __chgfdccsid(fd, (int)config->local_ccsid);
+  }
+#endif
+
   outs->s_isreg = TRUE;
   outs->fopened = TRUE;
   outs->stream = file;
@@ -308,6 +326,16 @@ size_t tool_write_cb(char *buffer, size_t sz, size_t nmemb, void *userdata)
   if(!outs->stream && !tool_create_output_file(outs, per->config))
     return CURL_WRITEFUNC_ERROR;
 
+#ifdef __MVS__
+  if(config->local_ccsid == 0 && outs->bytes == 0 && !config->resume_from && !getenv("_ENCODE_FILE_NEW")) {
+    int ccsid = __guess_ue(buffer, bytes, NULL, 0);
+    if(ccsid == 65535)
+      __chgfdccsid(fileno(outs->stream), FT_BINARY);
+    else
+      __chgfdccsid(fileno(outs->stream), ccsid);
+  }
+#endif
+
   if(is_tty && (outs->bytes < 2000) && !config->terminal_binary_ok) {
     /* binary output to terminal? */
     if(memchr(buffer, 0, bytes)) {
