diff --git i/src/tool_getparam.h w/src/tool_getparam.h
index 44eb361..7c785a4 100644
--- i/src/tool_getparam.h
+++ w/src/tool_getparam.h
@@ -270,6 +270,7 @@ typedef enum {
   C_STDERR,
   C_STYLED_OUTPUT,
   C_SUPPRESS_CONNECT_HEADERS,
+  C_TAG,
   C_TCP_FASTOPEN,
   C_TCP_NODELAY,
   C_TELNET_OPTION,
@@ -297,6 +298,7 @@ typedef enum {
   C_TRACE_TIME,
   C_IP_TOS,
   C_UNIX_SOCKET,
+  C_UPLOAD_ENCODING,
   C_UPLOAD_FILE,
   C_UPLOAD_FLAGS,
   C_URL,
