diff --git i/src/tool_cfgable.c w/src/tool_cfgable.c
index b907b24..d00226f 100644
--- i/src/tool_cfgable.c
+++ w/src/tool_cfgable.c
@@ -157,7 +157,9 @@ static void free_config_fields(struct OperationConfig *config)
   tool_safefree(config->krblevel);
   tool_safefree(config->oauth_bearer);
   tool_safefree(config->sasl_authzid);
+  tool_safefree(config->upload_encoding);
   tool_safefree(config->unix_socket_path);
+  tool_safefree(config->haproxy_clientip);
   tool_safefree(config->writeout);
   tool_safefree(config->proto_default);
 
