diff --git i/src/tool_getparam.c w/src/tool_getparam.c
index 841c6ff..65ffe6c 100644
--- i/src/tool_getparam.c
+++ w/src/tool_getparam.c
@@ -23,6 +23,11 @@
  ***************************************************************************/
 #include "tool_setup.h"
 
+#ifdef __MVS__
+#include <zos.h>
+#include <_Ccsid.h>
+#endif
+
 #include "tool_cfgable.h"
 #include "tool_cb_prg.h"
 #include "tool_filetime.h"
@@ -328,6 +333,7 @@ static const struct LongShort aliases[]= {
   {"stderr",                     ARG_FILE, ' ', C_STDERR},
   {"styled-output",              ARG_BOOL, ' ', C_STYLED_OUTPUT},
   {"suppress-connect-headers",   ARG_BOOL, ' ', C_SUPPRESS_CONNECT_HEADERS},
+  {"tag",                        ARG_STRG, ' ', C_TAG},
   {"tcp-fastopen",               ARG_BOOL, ' ', C_TCP_FASTOPEN},
   {"tcp-nodelay",                ARG_BOOL, ' ', C_TCP_NODELAY},
   {"telnet-option",              ARG_STRG, 't', C_TELNET_OPTION},
@@ -356,6 +362,7 @@ static const struct LongShort aliases[]= {
   {"trace-ids",                  ARG_BOOL, ' ', C_TRACE_IDS},
   {"trace-time",                 ARG_BOOL, ' ', C_TRACE_TIME},
   {"unix-socket",                ARG_FILE, ' ', C_UNIX_SOCKET},
+  {"upload-encoding",            ARG_STRG, ' ', C_UPLOAD_ENCODING},
   {"upload-file",                ARG_FILE, 'T', C_UPLOAD_FILE},
   {"upload-flags",               ARG_STRG, ' ', C_UPLOAD_FLAGS},
   {"url",                        ARG_STRG, ' ', C_URL},
@@ -2835,6 +2842,28 @@ static ParameterError opt_string(struct OperationConfig *config,
   case C_UPLOAD_FLAGS: /* --upload-flags */
     err = parse_upload_flags(config, nextarg);
     break;
+  case C_UPLOAD_ENCODING: /* --upload-encoding */
+    err = getstr(&config->upload_encoding, nextarg, DENY_BLANK);
+    break;
+  case C_TAG: /* --tag */
+#ifdef __MVS__
+    if(!strcasecmp(nextarg, "auto"))
+      config->local_ccsid = 0;
+    else if(!strcasecmp(nextarg, "binary"))
+      config->local_ccsid = 65535;
+    else {
+      unsigned short ccsid = __toCcsid((char*)nextarg);
+      if(ccsid == 0) {
+        warnf("Invalid --tag value: %s", nextarg);
+        err = PARAM_BAD_USE;
+      } else {
+        config->local_ccsid = ccsid;
+      }
+    }
+#else
+    warnf("--tag is ignored on this platform");
+#endif
+    break;
   }
   return err;
 }
